RiotSwiftUI/Modules/Room/PollHistory/Coordinator/PollHistoryCoordinator.swift

```diff
@@ -15,40 +15,91 @@
     private let pollHistoryHostingController: UIViewController
     private var pollHistoryViewModel: PollHistoryViewModelProtocol
+    private let navigationRouter: NavigationRouterType
     
     // Must be used only internally
     var childCoordinators: [Coordinator] = []
-    var completion: (() -> Void)?
+    var completion: ((MXEvent) -> Void)?
     
     init(parameters: PollHistoryCoordinatorParameters) {
         self.parameters = parameters
         let viewModel = PollHistoryViewModel(mode: parameters.mode, pollService: PollHistoryService(room: parameters.room, chunkSizeInDays: PollHistoryConstants.chunkSizeInDays))
         let view = PollHistory(viewModel: viewModel.context)
         pollHistoryViewModel = viewModel
         pollHistoryHostingController = VectorHostingController(rootView: view)
+        navigationRouter = parameters.navigationRouter
     }
     
     // MARK: - Public
     
     func start() {
         MXLog.debug("[PollHistoryCoordinator] did start.")
-        pollHistoryViewModel.completion = { _ in
-            
+        pollHistoryViewModel.completion = { [weak self] result in
+            switch result {
+            case .showPollDetail(let poll):
+                self?.showPollDetail(poll)
+            }
         }
     }
     
+    func showPollDetail(_ poll: TimelinePollDetails) {
+        
+        guard let event = parameters.room.mxSession.store.event(withEventId: poll.id, inRoom: parameters.room.roomId),
+              let detailCoordinator: PollHistoryDetailCoordinator = try? .init(parameters: .init(event: event, room: self.parameters.room)) else {
+            pollHistoryViewModel.context.alertInfo = .init(id: true, title: VectorL10n.settingsDiscoveryErrorMessage)
+            return
+        }
+        detailCoordinator.toPresentable().presentationController?.delegate = self
+        detailCoordinator.completion = { [weak self, weak detailCoordinator] result in
+            guard let self = self, let coordinator = detailCoordinator else { return }
+            switch result {
+            case .dismiss:
+                self.toPresentable().dismiss(animated: true)
+                self.remove(childCoordinator: coordinator)
+            case .viewInTimeline:
+                self.toPresentable().dismiss(animated: false)
+                self.remove(childCoordinator: coordinator)
+                var event = event
+                if poll.closed {
+                    let room = self.parameters.room
+                    let relatedEvents = room.mxSession.store.relations(forEvent: event.eventId, inRoom: room.roomId, relationType: MXEventRelationTypeReference)
+                    let pollEndedEvent = relatedEvents.first(where: { $0.eventType == .pollEnd })
+                    event = pollEndedEvent ?? event
+                }
+                self.completion?(event)
+            }
+        }
+        
+        self.add(childCoordinator: detailCoordinator)
```

Comment:
In general I'd prefer not to write `self.` wherever isn't really needed.
```suggestion
        add(childCoordinator: detailCoordinator)
```